### PR TITLE
jinx: post-install hooks and mapdb.json

### PR DESCRIPTION
**Pull request by sandersch**
Adds post-install hooks for spell-list.xml, gameobj-data.xml, and mapdb.json. They all reload the newly downloaded data file, and the mapdb hook also attempts to download any image files referenced in the mapdb that aren't already present in the map directory.

### CODE REVIEW

**Comment by sandersch** [comment]
If you want to test this before I merge https://github.com/elanthia-online/jinx/pull/45, you can 

```
;jinx repo add test https://deploy-preview-45--admiring-leakey-35fb01.netlify.app/
```

File: scripts/jinx.lic [hunk 1]
```diff
@@ -372,6 +372,18 @@ module Jinx
   end
 end
 
+module Jinx
+  module Lookup
+    class << self
+      def find_asset_in(repo, name)
+        repo[:available].find do |asset|
+          URI.decode(File.basename(asset[:file])).eql?(name)
```
Comment (mrhoribu) [inline scripts/jinx.lic:380]: `URI.decode` seems to be depreciated, at least in v2.7.2 of Ruby. I was able to change this to `URI.decode_www_form_component(File.basename(asset[:file])).eql?(name)` without it giving an undefined method.

File: scripts/jinx.lic [hunk 2]
```diff
@@ -533,10 +543,15 @@ module Jinx
       remote_name  = Installer.normalize_filename(remote_name)
       sources = sources.map {|source| Repo.manifest(source) }
       repo = Installer.ensure_specific(remote_name, sources)
-      asset = repo[:available].find {|remote_asset|
-        File.basename(remote_asset[:file]).eql?(remote_name)
-      }
-      local_asset_name = File.basename(local_asset_name || asset[:file])
+      asset = Jinx::Lookup.find_asset_in(repo, remote_name)
+      local_asset_name = URI.decode(File.basename(local_asset_name || asset[:file]))
```
Comment (mrhoribu) [inline scripts/jinx.lic:547]: Same here for depreciated URI.decode method. This should fix it: `local_asset_name = URI.decode_www_form_component(File.basename(local_asset_name || asset[:file]))`

**Review by mrhoribu — CHANGES_REQUESTED** [review]
Looks like URI.decode has been depreciated since v2.7.2 of Ruby. Had to update to new method name in order to have it work.

File: scripts/jinx.lic (hunk 1 above)
Comment (sandersch) [reply scripts/jinx.lic]: Updated this to use CGI.unescape which is available in Ruby 2.1 and 2.7 and doesn't have the same issue of trying to do too much in one method that made URI.unescape obsolete.

**Comment by sandersch** [comment]
> If you want to test this before I merge [elanthia-online/jinx#45](https://github.com/elanthia-online/jinx/pull/45), you can
> 
> ```
> ;jinx repo add test https://deploy-preview-45--admiring-leakey-35fb01.netlify.app/
> ```

I merged the mapdb and maps into the core repo so you should be able to 

```
;jinx data update mapdb.json --repo=core
```

with this version of ;jinx

File: scripts/jinx.lic (hunk 2 above)
Comment (mrhoribu) [reply scripts/jinx.lic:548]: Still need to update this to use CGI.escape as well. Then should be good.

Comment (sandersch) [reply scripts/jinx.lic:548]: Thanks, good catch. I missed this one at first but its updated now too.

**Review by mrhoribu — APPROVED** [review]

**Review by mrhoribu — APPROVED** [review]
Looks good to go! Thanks for doing this.

**Comment by ondreian** [comment]
shouldn't the map installer download a tarball from a jinx repo that away publishing can be more isolated for something like cartograph?

**Comment by sandersch** [comment]
Just trying to match the existing functionality of mapdb updating since ;repo is hard down. Open to ideas on how to improve this in the future but wanted to prioritize getting something simple working.

**Pull request merged**